domains/transfer/Transfer.API/API/Transfer/Api/v2023_11_23/Controllers/InternalTransferAgreementsController.cs

```diff
@@ -0,0 +1,31 @@
+using System.Linq;
+using System.Threading.Tasks;
+using API.Transfer.Api.Repository;
+using API.Transfer.Api.v2023_11_23.Dto.Responses;
+using Asp.Versioning;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace API.Transfer.Api.v2023_11_23.Controllers;
+
+[ApiExplorerSettings(IgnoreApi = true)]
+[ApiVersion("20231123")]
+[ApiController]
+public class InternalTransferAgreementsController(ITransferAgreementRepository agreementRepository)
+    : ControllerBase
+{
+    [HttpGet("api/internal-transfer-agreements/all")]
+    [AllowAnonymous]
+    public async Task<ActionResult> GetAll()
+    {
+        var transferAgreements = await agreementRepository.GetAllTransferAgreements();
+
+        return Ok(new InternalTransferAgreementsDto(transferAgreements.Select(ta => new InternalTransferAgreementDto(
```

Comment:
Why not just call them transfer-agreements?

Reply:
We already have a transfer agreement dto. This is ony to be used internally as it contains more IDs

Reply:
Not a fan of the name, but I can't come up with a better one, so I'll resolve